rackspace/autoscale/v1/policies/results.go

```diff
@@ -0,0 +1,138 @@
+package policies
+
+import (
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/pagination"
+)
+
+type policyResult struct {
+	gophercloud.Result
+}
+
+// Extract interprets any policyResult as a Policy, if possible.
+func (r policyResult) Extract() (*Policy, error) {
+	if r.Err != nil {
+		return nil, r.Err
+	}
+
+	var response struct {
+		Policy Policy `mapstructure:"policy"`
+	}
+
+	err := mapstructure.Decode(r.Body, &response)
+
+	return &response.Policy, err
+}
+
+// CreateResult represents the result of a create operation.
+type CreateResult struct {
+	policyResult
+}
+
+// Extract extracts a slice of Policies from a CreateResult.  Multiple policies
+// can be created in a single operation, so the result of a create is always a
+// list of policies.
+func (res CreateResult) Extract() ([]Policy, error) {
+	if res.Err != nil {
+		return nil, res.Err
+	}
+
+	return commonExtractPolicies(res.Body)
+}
+
+// GetResult temporarily contains the response from a Get call.
+type GetResult struct {
+	policyResult
+}
+
+// UpdateResult represents the result of an update operation.
+type UpdateResult struct {
+	gophercloud.ErrResult
+}
+
+// DeleteResult represents the result of a delete operation.
+type DeleteResult struct {
+	gophercloud.ErrResult
+}
+
+// ExecuteResult represents the result of an execute operation.
+type ExecuteResult struct {
+	gophercloud.ErrResult
+}
+
+// Policy represents a scaling policy.
+type Policy struct {
+	// UUID for the policy.
+	ID string `mapstructure:"id" json:"id"`
+
+	// Name of the policy.
+	Name string `mapstructure:"name" json:"name"`
+
+	// Type of scaling policy.
+	Type Type `mapstructure:"type" json:"type"`
+
+	// Cooldown period, in seconds.
+	Cooldown int `mapstructure:"cooldown" json:"cooldown"`
+
+	// Number of servers added or, if negative, removed.
+	Change interface{} `mapstructure:"change" json:"change"`
```

Comment:
`Change`, `ChangePercent`, and `DesiredCapacity` are empty interfaces here because a policy can only have one of the three. The API will return whichever field is appropriate and omit the others. In reality, `ChangePercent` is a float and the other two are integers.

So, this lets you test if each is `nil`, though everything comes back as a float64. I could add an `UnmarshalJSON` method that converts to integers as needed. Is that necessary? Or is this off base altogether?